packages/teuchos/CMakeLists.txt

```diff
@@ -226,19 +248,37 @@ TRIBITS_ADD_OPTION_AND_DEFINE(
   )
 
 IF (${PACKAGE_NAME}_ENABLE_COMPLEX)
+  #
+  # The check needs to run code, so give cross-compiling users a
+  # bypass.  This is not a complete check, but we have an error test
+  # below that explains the required settings.  (In the cross
+  # compilation case, running code here could get the wrong answer.
+  # Thus, it's better to error out than to continue.)
+  #
+  IF (DEFINED HAVE_COMPLEX_BLAS_PROBLEM)
+    MESSAGE(STATUS "You set HAVE_COMPLEX_BLAS_PROBLEM=${HAVE_COMPLEX_BLAS_PROBLEM}; this suggests you are cross compiling.  If you're not cross compiling, best practice is not to set this; Trilinos will figure it out for you.")
+  ELSE()
+    INCLUDE(CheckCXXComplexBlasProblem)
+    CHECK_CXX_COMPLEX_BLAS_PROBLEM(HAVE_COMPLEX_BLAS_PROBLEM)
 
-  INCLUDE(CheckCXXComplexBlasProblem)
-  CHECK_CXX_COMPLEX_BLAS_PROBLEM(HAVE_COMPLEX_BLAS_PROBLEM)
-
+    IF (HAVE_COMPLEX_BLAS_PROBLEM)
+      INCLUDE(CheckCXXComplexBlasProblemCanBeFixed)
+      IF(HAVE_TEUCHOS_BLASFLOAT_APPLE_VECLIB_BUGFIX)
+        CHECK_COMPLEX_BLAS_VECLIB_OKAY(HAVE_VECLIB_COMPLEX_BLAS)
+      ELSE()
+        CHECK_CXX_COMPLEX_BLAS_PROBLEM_CAN_BE_FIXED(HAVE_FIXABLE_COMPLEX_BLAS_PROBLEM)
+      ENDIF()
+    ENDIF()
+  ENDIF()
+  ASSERT_DEFINED(HAVE_COMPLEX_BLAS_PROBLEM)
   IF (HAVE_COMPLEX_BLAS_PROBLEM)
-    INCLUDE(CheckCXXComplexBlasProblemCanBeFixed)
-    IF(HAVE_TEUCHOS_BLASFLOAT_APPLE_VECLIB_BUGFIX)
-      CHECK_COMPLEX_BLAS_VECLIB_OKAY(HAVE_VECLIB_COMPLEX_BLAS)
-    ELSE()
-      CHECK_CXX_COMPLEX_BLAS_PROBLEM_CAN_BE_FIXED(HAVE_FIXABLE_COMPLEX_BLAS_PROBLEM)
+    IF ((NOT (DEFINED HAVE_VECLIB_COMPLEX_BLAS)) AND
+        (NOT (DEFINED HAVE_FIXABLE_COMPLEX_BLAS_PROBLEM)))
+      MESSAGE(FATAL_ERROR "IF HAVE_COMPLEX_BLAS_PROBLEM is defined and TRUE, then you must define at least one of HAVE_VECLIB_COMPLEX_BLAS or HAVE_FIXABLE_COMPLEX_BLAS_PROBLEM.  This is most likely to happen if you are cross compiling and setting these options yourself, rather than relying on CMake to set the options.")
     ENDIF()
   ENDIF()
-  GLOBAL_SET(HAVE_COMPLEX_BLAS 
+
+  GLOBAL_SET(HAVE_COMPLEX_BLAS
```

Comment:
I am not familiar with this type of CMake logic.  Does this work?  Can you assign a CMake variable to a boolen expression?

Reply:
@bartlettroscoe Not sure; that's what was there before.  I'm pretty sure that setting these to FALSE works, since that's what was happening in the cross-compilation case.